CoreFoundation/NumberDate.subproj/CFTimeZone.c

```diff
@@ -702,7 +829,9 @@ static CFTimeZoneRef __CFTimeZoneCreateSystem(void) {
         } else {
             name = CFStringCreateWithBytes(kCFAllocatorSystemDefault, (uint8_t *)linkbuf, strlen(linkbuf), kCFStringEncodingUTF8, false);
         }
-    } else {
+    } else
```

Comment:
I think it might be nicer to just end the previous block with the one duplicated line for the return and then avoid the `else` and braces for the remainder.

Reply:
I think it is not possible here, since there’s a block in the middle that has to be run for both branches. The return cannot be moved here.

Reply:
Hmm, did I miss that?  What is the block which needs to be executed?

Reply:
The `if (name)` block that starts in L890 (new version). The previous branches (the `if` and the `else` branch, only the `else` is valid for Android), are trying to find the `name`, while that last bit is find the information about the time zone with that name.

Reply:
How about folding `!TARGET_OS_ANDROID` into the `if` condition on L868?

Reply:
I'm not sure exactly why you mean (please provide the code of what you mean), but `__tzZoneInfo` is used in L866, and that variable doesn't exist in Android, so it has to be compiled out.